packages/ai/src/ui/chat.ts

```diff
@@ -93,11 +94,10 @@ export interface ChatState<UI_MESSAGE extends UIMessage> {
 
 export type ChatOnErrorCallback = (error: Error) => void;
 
-export type ChatOnToolCallCallback = ({
-  toolCall,
-}: {
-  toolCall: ToolCall<string, unknown>;
-}) => void | Promise<unknown> | unknown;
+export type ChatOnToolCallCallback<UI_MESSAGE extends UIMessage = UIMessage> =
```

Comment:
I've included the default value here to avoid breaking changes for users who might be using `ChatOnToolCallCallback`.